cs_upload.py

```diff
@@ -203,14 +203,26 @@ def upload_to_s3(self, form_data: str, trace_file: str) -> None:
         fields["success_action_status"] = str(fields["success_action_status"])
         fields["x-amz-meta-filename"] = basename(trace_file)
 
+        (base, outer_extension) = splitext(trace_file)
+        (_, inner_extension) = splitext(base)
+        if outer_extension == ".gz" and inner_extension == ".cst":
```

Comment:
Maybe `trace_file.endswith(".cst.gz")`?

Reply:
Yeah, in the end that's simpler, isn't it? I wanted to use the `os.path` tools but in this case they just make everything more complicated.